sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryAPITests.groovy

```diff
@@ -392,23 +396,91 @@ class DirectoryAPITests extends APISpec {
         }
         primaryDirectoryClient.createFile(dirPrefix + 2, 1024)
         for (int i = 0; i < 3; i++) {
-            nameList.add(dirPrefix + i)
+            nameList << (dirPrefix + i)
         }
 
         when:
-        def fileRefIter = primaryDirectoryClient.listFilesAndDirectories(namer.getResourcePrefix() + extraPrefix, maxResults, null, null).iterator()
+        def options = new ShareListFilesAndDirectoriesOptions()
+            .setPrefix(namer.getResourcePrefix() + extraPrefix)
+            .setMaxResultsPerPage(maxResults)
+            .setIncludeExtendedInfo(includeExtendedInfo) // set FIRST since subsequent can autoset this one
+            .setIncludeTimestamps(timestamps)
+            .setIncludeETag(etag)
+            .setIncludeAttributes(attributes)
+            .setIncludePermissionKey(permissionKey)
+        def returnedFileList = primaryDirectoryClient.listFilesAndDirectories(options, null, null).collect()
```

Comment:
let's keep tests with extended and without extended info separate. We'd like to run existing form of test for older service versions - if we mix them together we'll loose that ability.

You may need to annotate tests that take extended info with `@RequiredServiceVersion` annotation.